prog/dftb+/lib_dftb/densitymatrix.F90

```diff
@@ -71,20 +71,38 @@ subroutine fullDensityMatrix_real(dm, eigenvecs, filling)
     integer :: ii, nLevels
     real(dp) :: shift
 
+    !> detect and handle density matrix builds for non-Aufbau fillings
+    logical :: tNonAufbau = .false.
+    real(dp), allocatable :: tmpMtx(:)
+    integer :: mixIndx
+
     @:ASSERT(all(shape(eigenvecs) == shape(dm)))
     @:ASSERT(size(eigenvecs,dim=1) == size(eigenvecs,dim=2))
     @:ASSERT(size(eigenvecs,dim=1) == size(filling))
 
+
     dm(:,:) = 0.0_dp
     do ii =  size(filling), 1, -1
       nLevels = ii
       if (abs(filling(ii)) >= epsilon(1.0_dp)) then
         exit
       end if
     end do
+
+    ! determine whether fillings are non-Aufbau
+    mixIndx=nLevels-1 !sets to HOMO-1
```

Comment:
In 1 electron systems (H2+) this will cause problems below

Reply:
I can wrap that in a statement  "if (nLevels/=1) then..." to exclude one-electron systems in the ground state. The issue is testing a zero index in the filling vector correct?

Reply:
It might be worth working out a solution that could be used for 1 electron cases. I've made a PR against your branch that just uses the max() construct.

Reply:
If (H2+) were excited I think that nLevels==2, from what I can tell nLevels is determined by the first non-zero filling by reading the filling vector from the top. This little hack is only useful in saving the empty MO in the mixed state (the previous HOMO that has been excited from).